### ISSUE TITLE
Ability to update options on the fly, or exposing an API to re-initialize docsearch

### ISSUE THREAD

**Issue by Justineo** [algolia/docsearch#608]
**Do you want to request a _feature_ or report a _bug_?**

Feature request. Following up https://github.com/algolia/docsearch/issues/599#issuecomment-459970259.

My site uses a front-end router and I need docsearch to update options after user navigated to a new page from the front-end.

**What is the current behavior?**

The API only exposes an initialize function, lacking the ability to dispose or re-initialize from the front-end.

**If the current behavior is a bug, please provide all the steps to reproduce and a minimal
[JSFiddle][3] example or a repository on GitHub that we can `npm install`
and `npm start`.**

**What is the expected behavior?**

Docsearch can be re-initialized with different options, or setting new options on the fly.

[1]: https://github.com/algolia/docsearch-configs
[2]: https://community.algolia.com/docsearch/config-file.html
[3]: (https://jsfiddle.net/)

**Comment by Justineo** [comment]
After looking into the source code, I realized that it's already supported by modifying properties on the returned instance of the global `docsearch` API. I think it should be documented.

**Comment by s-pace** [comment]
Thanks for the update @Justineo 

Would you mind providing more details regarding what you wanted to achieve? If you can share with us a code example, we would be happy to include it within your documentation.

**Comment by Justineo** [comment]
@s-pace 

I'm using a front-end router. After user navigates to a new URL that triggers locale change, I need to update the `algoliaOptions` of the `docsearch` instance.

Something like:

```js
const instance = docsearch({ ... })

app.on('localechange', val => {
  instance.algoliaOptions.facetFilters = [`lang:${val}`]
})
```

**Comment by Justineo** [comment]
Furthermore, I found the `client.clearCache()` method doesn't cover all cases. If we follow such steps:

1. Search for `'a'`
2. Update `algoliaOptions`
3. Run `clearCache()`
4. Search for `'a'` again

We'll still get the cached results. The data is cached inside the `Dataset` object deep inside. After some digging, I finally managed to find a way to correctly clear all cached result:

```js
instance.autocomplete
  .eq(0).data('aaAutocomplete')
  .dropdown.datasets[0].clearCachedSuggestions()
instance.client.clearCache()
```
